core/identity-spi/src/main/java/software/amazon/awssdk/identity/spi/AwsCredentialsIdentity.java

```diff
@@ -45,54 +43,33 @@ public interface AwsCredentialsIdentity extends Identity {
      */
     String secretAccessKey();
 
+    static Builder builder() {
+        return DefaultAwsCredentialsIdentity.builder();
+    }
+
     /**
      * Constructs a new credentials object, with the specified AWS access key and AWS secret key.
      *
      * @param accessKeyId The AWS access key, used to identify the user interacting with services.
      * @param secretAccessKey The AWS secret access key, used to authenticate the user interacting with services.
      */
     static AwsCredentialsIdentity create(String accessKeyId, String secretAccessKey) {
-        Validate.paramNotNull(accessKeyId, "accessKeyId");
-        Validate.paramNotNull(secretAccessKey, "secretAccessKey");
-
-        return new AwsCredentialsIdentity() {
-            @Override
-            public String accessKeyId() {
-                return accessKeyId;
-            }
-
-            @Override
-            public String secretAccessKey() {
-                return secretAccessKey;
-            }
+        return builder().accessKeyId(accessKeyId)
+                        .secretAccessKey(secretAccessKey)
+                        .build();
+    }
 
-            @Override
-            public String toString() {
-                return ToString.builder("AwsCredentialsIdentity")
-                               .add("accessKeyId", accessKeyId)
-                               .build();
-            }
+    interface Builder {
```

Comment:
`extends CopyableBuilder`?

Reply:
I didn't think it was necessarily needed, but we can add it.